[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70320

Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

navahgar committed:
[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization



Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

[ghstack-poisoned]

eellison commented on torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp:
```diff
@@ -106,9 +106,55 @@ bool TryGeneralizeInputDimensionsToSymbolicShapes(
 bool GenerateGuard(Node* tensorexpr_graph_node, bool add_composed_op) {
   auto tensorexpr_graph = SubgraphUtils::getSubgraph(tensorexpr_graph_node);
 
+  // Move constant tensors from the subgraph to the outer scope.
```
Can you add a comment about why we're doing this ? Basically, Symbolic shape analysis doesn't currently handle well the broadcast(constant, symbolic_shape) case well

navahgar replied:
Done

navahgar committed:
Update on "[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization"


Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

[ghstack-poisoned]

navahgar committed:
Update on "[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization"


Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

[ghstack-poisoned]

navahgar committed:
Update on "[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization"


Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

[ghstack-poisoned]

navahgar committed:
Update on "[jit] [shape analysis] Move constant tensors out of fused subgraphs during generalization"


Differential Revision: [D33280508](https://our.internmc.facebook.com/intern/diff/D33280508/)

[ghstack-poisoned]
